rust/datafusion/src/datasource/datasource.rs

```diff
@@ -33,6 +33,14 @@ pub struct Statistics {
     pub num_rows: Option<usize>,
     /// total byte of the table rows
     pub total_byte_size: Option<usize>,
+    /// Statistics on a column level
+    pub column_statistics: Option<Vec<ColumnStatistics>>,
+}
+/// This table statistics are estimates about column
```

Comment:
Eventually the use of these statistics is probably more general than just datasources (aka for a cost based optimizer we would probably want the estimates to be attached to the output of all LogicalPlan nodes). 

But this is a good start for now!